Submodels/file_handling.py

```
@@ -162,18 +168,30 @@ def GetGridSize(self) -> list[float, float, float, float]:
             X_AFT = CONST_3 * MAX LENGTH (3?)
             """
 
+            # Validate inputs
+            if not isinstance(self.ducted_fan_design_params, dict):
+                raise TypeError("ducted_fan_design_params must be a dictionary!") from None
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)